dvc/repo/run.py

```diff
@@ -66,6 +70,35 @@ def run(
 
     with self.state:
         if not no_exec:
+            if (stage.cmd or stage.is_import) and not stage.locked:
+                existing = [
+                    out
+                    for out in stage.outs
+                    if os.path.exists(str(out))
+                    if out.changed_cache()
+                ]
+
+                statement = (
+                    "The following outputs will be removed to ensure that the"
+                    " command is creating them (enhancing reproducibility):\n"
+                    "{outs}\n"
+                    "Are you sure you want to proceed?".format(
+                        outs="\n".join("    - {}".format(x) for x in existing)
+                    )
+                )
+
+                if (
+                    existing
+                    and not remove_outs
+                    and not prompt.confirm(statement)
+                ):
+                    raise DvcException(
+                        "cannot destroy without a confirmation from the user."
+                        " Use --remove-outs if you don't want to be asked."
+                    )
+
+                stage.remove_outs(ignore_remove=False, force=False)
```

Comment:
I think we can move the logic inside remove_outs, add a flag and centralize the logic that calculates the list of files to remove

Reply:
@shcheklein , we have a `safe_remove` method already, but it handles a single path and the exception is suggesting the use of `--force` instead of `--remove-outs`, that's why I moved the logic outside. Also, the `force` argument in `remove_outs` is used for removing _persisted outputs_, maybe adding a `prompt=False` argument? I'll try different stuff